scheduler/scheduler-server/src/test/java/performancetests/metrics/TaskSchedulingTimeTest.java

```diff
@@ -0,0 +1,168 @@
+
+        Set<Long> ids = new HashSet<>();
+        for (JobTaskStatusListener.TimestampedData<TaskInfo> tData : listener.getTaskEvents()) {
+            if (tData.getData().getJobId().equals(jobId) && tData.getData().getStatus().equals(TaskStatus.RUNNING)) {
+                ids.add(tData.getData().getTaskId().longValue());
+            }
+        }
+
+        assertEquals(taskNumber, ids.size());
+
+        for (JobTaskStatusListener.TimestampedData<TaskInfo> tData : listener.getTaskEvents()) {
+            if (tData.getData().getJobId().equals(jobId) && (tData.getData().getStatus().equals(TaskStatus.ABORTED) ||
+                    tData.getData().getStatus().equals(TaskStatus.FAILED) ||
+                    tData.getData().getStatus().equals(TaskStatus.FAULTY) ||
+                    tData.getData().getStatus().equals(TaskStatus.FINISHED) ||
+                    tData.getData().getStatus().equals(TaskStatus.PAUSED) ||
+                    tData.getData().getStatus().equals(TaskStatus.PENDING))) {
+                throw new RuntimeException("There should not be task with any of these statuses.");
+            }
+        }
+
+        List<Long> longList = new ArrayList<>();
+
+        for (JobTaskStatusListener.TimestampedData<TaskInfo> tData : listener.getTaskEvents()) {
+            if (tData.getData().getJobId().equals(jobId) && tData.getData().getStatus().equals(TaskStatus.RUNNING)) {
+                longList.add(tData.getTimestamp());
+            }
+        }
+
+        Collections.sort(longList);
+
+        long first = longList.get(0);
+        long last = longList.get(longList.size() - 1);
+
+        long anActualTime = last - first;
+
+        assertTrue(last > first);
+
+        LOGGER.info(makeCSVString(TaskSchedulingTimeTest.class.getSimpleName(),
+                taskNumber,
+                timeLimit,
+                anActualTime,
+                ((anActualTime < timeLimit) ? SUCCESS : FAILURE)));
+
+        assertThat(String.format("Task creation rate for job with %s tasks", taskNumber),
+                anActualTime,
+                lessThan(timeLimit));
+
+    }
+
+    private long waitUntilNumberOfTasksEvents(JobTaskStatusListener listener, JobId jobId, TaskStatus taskStatus,
+                                              int numberOfEvents) throws InterruptedException {
+        long result = 0;
+
+        while (result < numberOfEvents) {
+            result = 0;
+            for (JobTaskStatusListener.TimestampedData<TaskInfo> tData : listener.getTaskEvents()) {
+                if (tData.getData().getJobId().equals(jobId) && tData.getData().getStatus().equals(taskStatus)) {
+                    ++result;
+                }
+            }
+
+            if (result > numberOfEvents) {
+                throw new RuntimeException(String.format("There are more %s tasks %d that we expected",
+                        taskStatus.toString(),
+                        result));
+            }
+            Thread.sleep(1000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)